src/utilities/include/tri_dist.h

```diff
@@ -0,0 +1,262 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <array>
+#include <glm/glm.hpp>
+
+namespace manifold {
+
+template <typename T>
+const T& clamp(const T& v, const T& lo, const T& hi) {
+  assert(!(hi < lo));
+  return (v < lo) ? lo : (hi < v) ? hi : v;
+}
+
+// From NVIDIA-Omniverse PhysX - BSD 3-Clause "New" or "Revised" License
+// https://github.com/NVIDIA-Omniverse/PhysX/blob/main/LICENSE.md
+// https://github.com/NVIDIA-Omniverse/PhysX/blob/main/physx/source/geomutils/src/sweep/GuSweepCapsuleCapsule.cpp
+// With minor modifications to use glm::vec3 type.
+
+/**
+ * Returns the distance between two line segments.
+ *
+ * @param[out] cq Closest point on line segment pa.
+ * @param[out] cp Closest point on line segment qb.
+ * @param[in]  p  One endpoint of the first line segment.
+ * @param[in]  q  Other endpoint of the first line segment.
+ * @param[in]  p  One endpoint of the second line segment.
+ * @param[in]  q  Other endpoint of the second line segment.
+ */
+void EdgeEdgeDist(glm::vec3& x, glm::vec3& y,  // closest points
+                  const glm::vec3& p,
+                  const glm::vec3& a,  // seg 1 origin, vector
+                  const glm::vec3& q,
+                  const glm::vec3& b)  // seg 2 origin, vector
+{
+  const glm::vec3 T = q - p;
+  const double ADotA = glm::dot(a, a);
+  const double BDotB = glm::dot(b, b);
+  const double ADotB = glm::dot(a, b);
+  const double ADotT = glm::dot(a, T);
+  const double BDotT = glm::dot(b, T);
+
+  // t parameterizes ray (p, a)
+  // u parameterizes ray (q, b)
+
+  // Compute t for the closest point on ray (p, a) to ray (q, b)
+  const double Denom = ADotA * BDotB - ADotB * ADotB;
+
+  double t;  // We will clamp result so t is on the segment (p, a)
+  if (Denom != 0.0f)
+    t = clamp((ADotT * BDotB - BDotT * ADotB) / Denom, 0.0, 1.0);
+  else
+    t = 0.0f;
+
+  // find u for point on ray (q, b) closest to point at t
+  double u;
+  if (BDotB != 0.0f) {
+    u = (t * ADotB - BDotT) / BDotB;
+
+    // if u is on segment (q, b), t and u correspond to closest points,
+    // otherwise, clamp u, recompute and clamp t
+    if (u < 0.0f) {
+      u = 0.0f;
+      if (ADotA != 0.0f)
+        t = clamp(ADotT / ADotA, 0.0, 1.0);
+      else
+        t = 0.0f;
```

Comment:
Let's convert these to ternaries where possible - makes the code cleaner in my opinion.

Reply:
Done!